compiler/src/dotty/tools/dotc/typer/EtaExpansion.scala

```diff
@@ -46,7 +46,8 @@ abstract class Lifter {
     if (noLift(expr)) expr
     else {
       val name = UniqueName.fresh(prefix)
-      var liftedType = fullyDefinedType(expr.tpe.widen, "lifted expression", expr.pos)
+      // don't instantiate here, as the type params could be further constrained, see tests/pos/pickleinf.scala
```

Comment:
After looking into it this _might_ be OK because we're more careful about how we deal with signatures containing type variables in recent versions of Dotty, I've found one related issue for which I've opened a PR: https://github.com/lampepfl/dotty/pull/4593

Reply:
Thanks a lot for looking into it @smarter !